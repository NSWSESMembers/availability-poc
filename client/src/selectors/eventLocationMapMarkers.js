/*
  Give it an array of event location objects and it will return and array of
  react-native-maps/map-marker friendly objects (with some extra data mixed in)
*/

// eslint-disable-next-line import/prefer-default-export
export const eventLocationMapMarkers = (eventLocations) => {
  const mapMarkers = [];
  eventLocations.forEach((r) => {
    if (r.locationLatitude !== null && r.locationLongitude !== null) {
      mapMarkers.push({
        id: r.name,
        latitude: r.locationLatitude,
        longitude: r.locationLongitude,
        icon: r.icon,
      });
    }
  });
  return mapMarkers;
};
