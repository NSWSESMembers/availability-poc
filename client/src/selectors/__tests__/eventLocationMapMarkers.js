import { eventLocationMapMarkers } from '../eventLocationMapMarkers';

const aMarker = [{
  name: 'markera',
  locationLatitude: 0,
  locationLongitude: 0,
  icon: 'icon',
}];

const bMarker = [{
  name: 'markerb',
  locationLatitude: 0,
  locationLongitude: 0,
  icon: 'icon',
}];

test('user marker', () => {
  expect(
    eventLocationMapMarkers(aMarker),
  ).toMatchSnapshot();
  expect(
    eventLocationMapMarkers(aMarker.concat(bMarker)),
  ).toMatchSnapshot();
});
