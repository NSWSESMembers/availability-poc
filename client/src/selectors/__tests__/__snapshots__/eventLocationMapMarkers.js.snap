// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user marker 1`] = `
Array [
  Object {
    "icon": "icon",
    "id": "markera",
    "latitude": 0,
    "longitude": 0,
  },
]
`;

exports[`user marker 2`] = `
Array [
  Object {
    "icon": "icon",
    "id": "markera",
    "latitude": 0,
    "longitude": 0,
  },
  Object {
    "icon": "icon",
    "id": "markerb",
    "latitude": 0,
    "longitude": 0,
  },
]
`;
